Code/Framework/AzCore/AzCore/Name/Name.h

```diff
@@ -82,62 +114,101 @@ namespace AZ
             return GetHash() != other.GetHash();
         }
 
-        // We delete these operators because using Name in ordered containers is not supported. 
+        bool operator==(const NameRef& other) const
+        {
+            return m_data == other;
+        }
+
+        bool operator!=(const NameRef& other) const
+        {
+            return m_data != other;
+        }
+
+        // We delete these operators because using Name in ordered containers is not supported.
         // The point of Name is for fast equality comparison and lookup. Unordered containers should be used instead.
         friend bool operator<(const Name& lhs, const Name& rhs) = delete;
         friend bool operator<=(const Name& lhs, const Name& rhs) = delete;
         friend bool operator>(const Name& lhs, const Name& rhs) = delete;
         friend bool operator>=(const Name& lhs, const Name& rhs) = delete;
-        
+
         //! Returns the string's hash that is used as the key in the NameDictionary.
         Hash GetHash() const
         {
             return m_hash;
         }
 
+        AZ_FORCE_INLINE static Name* GetDeferredHead()
+        {
+            return s_staticNameBegin;
+        }
```

Comment:
The `AZ_FORCE_INLINE` isn't needed here.
The function is inline to the declaration, so there is no need of the `inline` keyword either.

Furthermore I believe you need to return this by reference.
```suggestion
        static Name*& GetDeferredHead()
        {
            return s_staticNameBegin;
        }
```

Reply:
It shouldn't be needed to return it by ref (we never update this deferred head, we leave it intact) but if we ever decide we want to clear the module-level deferred head it might be a good idea, made the update.